swipe-reveal-layout/src/main/java/com/chauthai/swipereveallayout/SwipeRevealLayout.java

```diff
@@ -355,37 +355,54 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         int desiredWidth = 0;
         int desiredHeight = 0;
 
+        int wrapChildHeight = 0;
+        int wrapChildWidth = 0;
+
         // first find the largest child
         for (int i = 0; i < getChildCount(); i++) {
             final View child = getChildAt(i);
+            final LayoutParams childParams = child.getLayoutParams();
             measureChild(child, widthMeasureSpec, heightMeasureSpec);
-            desiredWidth = Math.max(child.getMeasuredWidth(), desiredWidth);
-            desiredHeight = Math.max(child.getMeasuredHeight(), desiredHeight);
+
+            if (childParams.height == LayoutParams.WRAP_CONTENT &&
+                    (mDragEdge == DRAG_EDGE_LEFT || mDragEdge == DRAG_EDGE_RIGHT))
+                wrapChildHeight = Math.max(child.getMeasuredHeight(), wrapChildHeight);
+            else
+                desiredHeight = Math.max(child.getMeasuredHeight(), desiredHeight);
+
+            if (childParams.width == LayoutParams.WRAP_CONTENT &&
+                    (mDragEdge == DRAG_EDGE_TOP || mDragEdge == DRAG_EDGE_BOTTOM))
+                wrapChildWidth = Math.max(child.getMeasuredWidth(), wrapChildWidth);
+            else
+                desiredWidth = Math.max(child.getMeasuredWidth(), desiredWidth);
+
         }
+        if (wrapChildHeight != 0)
+            desiredHeight = wrapChildHeight;
+        if (wrapChildWidth != 0)
+            desiredWidth = wrapChildWidth;
+
         // create new measure spec using the largest child width
         widthMeasureSpec = MeasureSpec.makeMeasureSpec(desiredWidth, widthMode);
-        heightMeasureSpec = MeasureSpec.makeMeasureSpec(desiredHeight, heightMode);
+        heightMeasureSpec = MeasureSpec.makeMeasureSpec(desiredHeight, MeasureSpec.EXACTLY);
```

Comment:
@mortenholmgaard  In our case, we had to keep the current `heightMode` to have the correct behavior. If we use `MeasureSpec.EXACTLY` it stretches too much the view : we loose some margins.
And thanks for the fix by the way !